unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ExploreHUD/Scripts/Handlers/ThumbnailHandler.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using DCL;
+using UnityEngine;
+
+internal class ThumbnailHandler : IDisposable
+{
+    public Texture2D texture { private set; get; }
+
+    AssetPromise_Texture texturePromise = null;
+
+    public void FetchThumbnail(string url, Action<Texture2D> onSuccess, Action onFail)
+    {
+        if (!(texture is null))
+        {
+            onSuccess?.Invoke(texture);
+        }
+        else if (string.IsNullOrEmpty(url))
+        {
+            onFail?.Invoke();
+        }
+        else if (texturePromise is null)
+        {
+            texturePromise = new AssetPromise_Texture(url, storeTexAsNonReadable: false);
+            texturePromise.OnSuccessEvent += textureAsset =>
+            {
+                texture = textureAsset.texture;
+                onSuccess?.Invoke(texture);
+            };
+            texturePromise.OnFailEvent += textureAsset =>
+            {
+                texturePromise = null;
+                onFail?.Invoke();
+            };
+            AssetPromiseKeeper_Texture.i.Keep(texturePromise);
+        }
+    }
+
+    public void Dispose()
+    {
+        if (texture)
+        {
+            GameObject.Destroy(texture);
+        }
+
+        if (texturePromise != null)
+        {
+            AssetPromiseKeeper_Texture.i.Forget(texturePromise);
+            texturePromise = null;
+        }
```

Comment:
we should do either one of there 2 things, not both. In the current state of this code we could be destroying a texture that still has references through the promise.

So I'd suggest checking for the texture promise existence first, and doing the `Forget` in that case, and if the promise is null, then in that case we can destroy the texture in this script